### ISSUE TITLE
Able to add invalid environment variables

### ISSUE THREAD

**Issue by weyert** [dokku/dokku#3008]
Description of problem:
A problem exists in Dokku were you are able to add a invalid environment variable through `dokku config:set` which accepts this invalid variable only when you try to delete it by unsetting it. The environment variable is not correctly deleted from the `ENV`-file needs manual assistance. 

Output of the following commands
```
-----> uname: Linux scw-c4bcc1 4.10.8-docker-1 #1 SMP Tue Apr 4 16:18:07 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
-----> memory: 
              total        used        free      shared  buff/cache   available
Mem:           2001         450         494          49        1057        1203
Swap:             0           0           0
-----> docker version: 
Client:
 Version:      17.11.0-ce
 API version:  1.34
 Go version:   go1.8.3
 Git commit:   1caf76c
 Built:        Mon Nov 20 18:37:39 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.11.0-ce
 API version:  1.34 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   1caf76c
 Built:        Mon Nov 20 18:36:09 2017
 OS/Arch:      linux/amd64
 Experimental: false
-----> docker daemon info: 
Containers: 16
 Running: 12
 Paused: 0
 Stopped: 4
Images: 53
Server Version: 17.11.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 992280e8e265f491f7a624ab82f3e238be086e49
runc version: 0351df1c5a66838d0c392b4ac4cf9450de844e2d
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.10.8-docker-1
Operating System: Ubuntu 16.04.3 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.955GiB
Name: scw-c4bcc1
ID: OKEA:4J26:JSQZ:4WQR:5AMJ:STB3:CWW4:TYMW:FWAV:TB5G:4LXX:MSUZ
Docker Root Dir: /var/lib/docker
Debug Mode (client): true
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

-----> sigil version: 0.4.0
-----> herokuish version: 
herokuish: 0.3.33
buildpacks:
  heroku-buildpack-multi     v1.0.0
  heroku-buildpack-ruby      v170
  heroku-buildpack-nodejs    v111
  heroku-buildpack-clojure   v77
  heroku-buildpack-python    v120
  heroku-buildpack-java      v57
  heroku-buildpack-gradle    v24
  heroku-buildpack-scala     v79
  heroku-buildpack-play      v26
  heroku-buildpack-php       v126
  heroku-buildpack-go        v78
  heroku-buildpack-erlang    fa17af9
  buildpack-nginx            v8
-----> dokku version: 0.11.0
-----> dokku plugins: 
plugn: 0.3.0
  00_dokku-standard    0.11.0 enabled    dokku core standard plugin
  20_events            0.11.0 enabled    dokku core events logging plugin
  apps                 0.11.0 enabled    dokku core apps plugin
  build-env            0.11.0 enabled    dokku core build-env plugin
  certs                0.11.0 enabled    dokku core certificate management plugin
  checks               0.11.0 enabled    dokku core checks plugin
  common               0.11.0 enabled    dokku core common plugin
  config               0.11.0 enabled    dokku core config plugin
  docker-options       0.11.0 enabled    dokku core docker-options plugin
  domains              0.11.0 enabled    dokku core domains plugin
  elasticsearch        1.0.0 enabled    dokku elasticsearch service plugin
  enter                0.11.0 enabled    dokku core enter plugin
  git                  0.11.0 enabled    dokku core git plugin
  haproxy              0.1.1 enabled    haproxy tcp load balancer for dokku
  letsencrypt          0.8.6 enabled    Automated installation of let's encrypt TLS certificates
  logs                 0.11.0 enabled    dokku core logs plugin
  maintenance          0.5.0 enabled    Maintenance mode for apps
  named-containers     0.11.0 enabled    dokku core named containers plugin
  network              0.11.0 enabled    dokku core network plugin
  nginx-vhosts         0.11.0 enabled    dokku core nginx-vhosts plugin
  plugin               0.11.0 enabled    dokku core plugin plugin
  postgres             1.0.0 enabled    dokku postgres service plugin
  proxy                0.11.0 enabled    dokku core proxy plugin
  ps                   0.11.0 enabled    dokku core ps plugin
  rabbitmq             1.0.0 enabled    dokku rabbitmq service plugin
  redis                1.0.0 enabled    dokku redis service plugin
  repo                 0.11.0 enabled    dokku core repo plugin
  shell                0.11.0 enabled    dokku core shell plugin
  ssh-keys             0.11.0 enabled    dokku core ssh-keys plugin
  storage              0.11.0 enabled    dokku core storage plugin
  tags                 0.11.0 enabled    dokku core tags plugin
  tar                  0.11.0 enabled    dokku core tar plugin
```
Environment details (AWS, VirtualBox, physical, etc.):
VPS Scaleway

How was Dokku installed?:
Through the shell script described on the Dokku homepage

How reproducible:

Steps to Reproduce:
1. Add a invalid environment variable: `dokku config:set APP_NAME postgresql://postgres:postgres@example.com:5432/dbname?pool=25`

```
root@scw-c4bcc1:~# dokku config:set APP-NAME postgresql://postgres:postgres@example.com:5432/dbname?pool=25
-----> Setting config vars
       postgresql://postgres:postgres@example.com:5432/dbname?pool:  25
-----> Restarting app APP-NAME
```
2. `dokku config APP_NAME` notice the variable is added
3.  Remove the variable via `dokku config:unset APP_NAME postgresql://postgres:postgres@example.com:5432/dbname?pool=25`

```
root@scw-c4bcc1:~# dokku config:unset APP_NAME postgresql://postgres:postgres@example.com:5432/dbname?pool
-----> Unsetting postgresql://postgres:postgres@example.com:5432/dbname?pool
-----> Restarting app APP_NAME
```
4. Notice that the variable is available: 
Now something similar to below should listed in your `ENV` file:
`postgresql:            //postgres:postgres@example.com:5432/dbname?pool="25"`

Actual Results:
The invalid environment variable is added and then later not correctly deleted

Expected Results:
I would expect that the command doesn't accept these invalid environment variables to avoid this issue

**Comment by josegonzalez** [comment]
@alexquick any chance you can look into this? I'm guessing we just aren't validating env vars during `config:set`.

**Comment by alexquick** [comment]
@josegonzalez I'll look into it later tonight. We should definitely prevent invalid env vars from
 being added. I have a feeling this is revealing subtle bug in how adding vs removing vars is handled as well so I will also try to track that down.

**Comment by alexquick** [comment]
@josegonzalez there are a few things going on here:
1) If you `config:unset` a key that doesn't exist it silently fails. I believe that's how the old config plugin worked as well though I'm not entirely certain. I'll change the log message to mention that the key was not set in the first place. Do you want me to add a nonzero exit code in this case as well?

2) joho/godotenv has yaml support which is why this key is messing things up. We are willing to restrict our valid env var names to be valid bash identifiers, right? Env var names in a "setenv" sense can contain nearly anything but `=`.

**Comment by josegonzalez** [comment]
1. Continue to exit 0.
2. Yeah, lets restrict to valid bash identifiers. For reference, [here](https://github.com/dokku/dokku/blob/ef341a4fd609b5fdbc35c0250c73f8ff5a66cc5c/plugins/config/functions#L158) is an example for how we matched env var keys to values.

In testing this, I noticed that the app restart is silent when invoked via the config plugin. Don't know how hard that would be to fix, but would be nice to have all the output written to stdout/stderr during the restart invokation.
